### PR TITLE
[SHARED DEV] [#162195] Update permissions so admins can perform actions

### PR DESCRIPTION
**Pull request by LeticiaErrandonea**
# Release Notes
- Update permissions so admins can perform actions on Orders.

# Screenshot
## Actions
image

# Additional Context

# Accessibility
- [ ] Did you scan for accessibility issues?
- [ ] Did you check our accessibility goal checklist?
- [ ] Did you add accessibility [specs](https://github.com/dequelabs/axe-core-gems/blob/develop/packages/axe-core-rspec/README.md)?

### CODE REVIEW

File: app/forms/add_to_order_form.rb [hunk 1]
```diff
@@ -227,19 +220,26 @@ def merge_order
     return @merge_order if defined?(@merge_order)
 
     products = product.is_a?(Bundle) ? product.products : [product]
+    order = facility_order_in_project || original_order
```
Comment (LeticiaErrandonea) [inline app/forms/add_to_order_form.rb:223]: If there's a project, also add the `cross_core_project` to the new Order

File: app/controllers/reservations_controller.rb [hunk 2]
```diff
@@ -412,7 +412,10 @@ def original_project_order
     return @original_project_order if defined?(@original_project_order)
 
     project = @order.cross_core_project
-    @original_project_order = project.orders.find { |order| order.facility_id == project.facility_id }
+
+    @original_project_order = if project.present?
```
Comment (LeticiaErrandonea) [inline app/controllers/reservations_controller.rb:417]: Prefer `return if project.blank?`

File: app/lib/ability.rb [hunk 3]
```diff
@@ -177,10 +177,39 @@ def facility_administrator_abilities(user, resource, controller)
...
+        can :manage, OrderDetail, order: { facility_id: resource.order.facility_id }
+      end
+
+      if SettingsHelper.feature_on?(:cross_core_projects)
+        project = resource.order.cross_core_project
+
+        if project.present?
+          original_order = project.orders.first
+
+          can [:add_accessories, :new, :show, :update, :cancel], OrderDetail if user.facility_administrator_of?(original_order.facility)
```
Comment (LeticiaErrandonea) [inline app/lib/ability.rb:210]:
```suggestion
          can [:add_accessories, :new, :show, :update, :cancel], OrderDetail if user.facility_administrator_of?(project.facility)
```

File: app/controllers/reservations_controller.rb [hunk 4]
```diff
@@ -136,7 +142,7 @@ def new
 
     authorize! :new, @reservation
 
-    unless @instrument.can_be_used_by?(@order_detail.user)
+    unless @instrument.can_be_used_by?(@order_detail.user) || (@order_detail.order.cross_core_project.present? && @instrument.can_be_used_by?(@order_detail.order.created_by_user))
```
Comment (LeticiaErrandonea) [inline app/controllers/reservations_controller.rb:145]:
```suggestion
    can_use_instrument = @order_detail.order.cross_core_project.present? ? @instrument.can_be_used_by?(@order_detail.order.created_by_user) : @instrument.can_be_used_by?(@order_detail.user)
    unless can_use_instrument
```

File: app/controllers/facility_orders_controller.rb [hunk 5]
```diff
@@ -98,9 +98,20 @@ def load_order
   end
 
   def load_merge_orders
-    project_order_ids = SettingsHelper.feature_on?(:cross_core_projects) && @order.cross_core_project_id.present? ? @order.cross_core_project.orders.pluck(:id) : @order.id
+    missing_reservation_order_ids = []
```
[the remark this answers is not in the document]
Comment (LeticiaErrandonea) [reply app/controllers/facility_orders_controller.rb:101]: You mean setting `missing_reservation_order_ids` and `project_order_ids` based on a new method's result?

[the remark this answers is not in the document]
Comment (giladshanan) [reply app/controllers/facility_orders_controller.rb:101]: I was thinking something like:
```
@merge_orders = if missing_reservation_order_ids.any?
  Order.where(merge_with_order_id: project_order_ids, created_by: current_user.id).or(Order.where(id: missing_reservation_order_ids))
else
  Order.where(merge_with_order_id: project_order_ids, created_by: current_user.id)
end
```

File: app/controllers/facility_order_details_controller.rb [hunk 6]
```diff
@@ -43,9 +43,18 @@ def destroy
       end
     else
       flash[:notice] = I18n.t "controllers.facility_order_details.destroy.notice"
+
+      if @order_detail.order.cross_core_project.present?
+        return redirect_back(fallback_location: root_path)
+      end
+
       return redirect_to facility_order_path(current_facility, @order)
```
[the remark this answers is not in the document]
Comment (giladshanan) [reply app/controllers/facility_order_details_controller.rb:51]:
```
    if @order_detail.order.cross_core_project.present?
      redirect_back(fallback_location: root_path)
    else
      redirect_to(facility_order_path(current_facility, redirect_to_order))
    end
```

File: app/forms/add_to_order_form.rb [hunk 7]
```diff
@@ -227,19 +220,27 @@ def merge_order
...
                      Order.create!(
-                       merge_with_order_id: original_order.id,
-                       facility_id: original_order.facility_id,
+                       merge_with_order_id: target_order.id,
+                       facility_id: target_order.facility_id,
                        account_id: account_id,
-                       user_id: original_order.user_id,
+                       user_id: target_order.user_id,
                        created_by: created_by.id,
+                       cross_core_project: @order_project,
```
Comment (giladshanan) [inline app/forms/add_to_order_form.rb:231]: Can you add a spec for this?

**Review by giladshanan — COMMENTED** [review]
Just a couple more things!

File: app/controllers/facility_orders_controller.rb [hunk 8]
```diff
@@ -149,4 +149,11 @@ def load_cross_core_order_details
     end
   end
 
+  def missing_reservation_order_ids(project_order_ids)
+    return [] if project_order_ids.empty?
+
+    orders_with_only_one_order_detail = @order.cross_core_project.orders.select { |order| order.order_details.count == 1 }
```
Comment (giladshanan) [inline app/controllers/facility_orders_controller.rb:155]: we don't need to check `if project_order_ids.length > 1`?

**Review by giladshanan — APPROVED** [review]
🎉

**Pull request merged**